data_chunk.go

```diff
@@ -58,19 +58,18 @@ func (chunk *DataChunk) SetValue(colIdx int, rowIdx int, val any) error {
 	}
 	column := &chunk.columns[colIdx]
 
-	// Ensure that the types match before attempting to set anything.
-	// This is done to prevent failures 'halfway through' writing column values,
-	// potentially corrupting data in that column.
-	// FIXME: Can we improve efficiency here? We are casting back-and-forth to any A LOT.
-	// FIXME: Maybe we can make columnar insertions unsafe, i.e., we always assume a correct type.
-	v, err := column.tryCast(val)
-	if err != nil {
-		return addIndexToError(err, colIdx)
-	}
-
 	// Set the value.
-	column.setFn(column, C.idx_t(rowIdx), v)
-	return nil
+	return column.setFn(column, C.idx_t(rowIdx), val)
+}
+
+// SetValue writes a single value to a column in a data chunk.
+// Note that this requires casting the type for each invocation.
+// NOTE: Custom ENUM types must be passed as string.
+func SetChunkValue[T any](chunk DataChunk, colIdx int, rowIdx int, val T) error {
```

Comment:
I changed the documentation, but I don't think we should name it `TypedXxXx` because I don't think being typed, warrants being represented in the name. The current name is simple, concise, and descriptive of what the function does

Reply:
Fair point 👍